### PR TITLE
Feature/inventory

### LINKED ISSUE #13: 3.3 Inventory (7h)

### LINKED ISSUE #14: 3.3.1 Create/Update an inventory item (3h)

### LINKED ISSUE #15: 3.3.2 Remove an inventory item (1h)

### LINKED ISSUE #16: 3.3.3 Display items (3h)

### PR DESCRIPTION
**Pull request by VanHoanTran**
Closes #13 
Closes #14 
Closes #15 
Closes #16

### CODE REVIEW

**Comment by ghost** [comment]
Hey @VanHoanTran Could you please
- git fetch
- git merge origin/develop

resolve the conflicts, add everything to "staged"
- git merge --continue
- git push

Please?
Your branch has a bunch of conflicts compared to `develop` because a lot has been changed there since this branch was opened

Let me know if you need help with this

**Comment by VanHoanTran** [comment]
> Hey @VanHoanTran Could you please
> 
> * git fetch
> * git merge origin/develop
> 
> resolve the conflicts, add everything to "staged"
> 
> * git merge --continue
> * git push
> 
> Please? Your branch has a bunch of conflicts compared to `develop` because a lot has been changed there since this branch was opened
> 
> Let me know if you need help with this

@dadolhay, I already solved conflicts. please check it out. thank you

File: packages/mechanic-shop-fe/src/Routes.js [hunk 1]
```diff
@@ -28,6 +31,20 @@ const Routes = () => {
             path={paths.dashboard().pathname}
             element={<ShopDashboardScreen />}
           />
+          <Route
+            path={paths.inventory().pathname}
+            element={<InventoryScreen />}
```
Comment (ghost) [inline packages/mechanic-shop-fe/src/Routes.js]: We call these "parts"
(we merged inventory into parts at the end of last term)

So please use names like:
-PartList / PartListScreen
-EditPartForm

-EditPartScreen (can use EditPartForm)
-AddPartScreen (can also use EditPartForm)

**Review by ghost — CHANGES_REQUESTED** [review]
Only some minor naming changes

Amazing work!!

File: packages/mechanic-shop-fe/src/Routes.js (hunk 1 above)
Comment (VanHoanTran) [reply packages/mechanic-shop-fe/src/Routes.js]: I changed all the pathnames and components. Let me know if anything needs to be changed. I have not completed how to display a single part in the addPartForm. I will do it as soon as I can

File: packages/mechanic-shop-fe/src/Routes.js [hunk 2]
```diff
@@ -28,6 +31,20 @@ const Routes = () => {
             path={paths.dashboard().pathname}
             element={<ShopDashboardScreen />}
           />
+          <Route
+            path={paths.partList().pathname}
+            element={<PartListScreen />}
+          />
+
+          <Route
+            path={paths.addPartForm().pathname}
```
Comment (ghost) [inline packages/mechanic-shop-fe/src/Routes.js]: Please call this addPart

File: packages/mechanic-shop-fe/src/Routes.js [hunk 3]
```diff
@@ -28,6 +31,20 @@ const Routes = () => {
...
+            element={<AddPartScreen />}
+          />
+
+          <Route
+            path={paths.editPartForm().pathname}
```
Comment (ghost) [inline packages/mechanic-shop-fe/src/Routes.js]: Please call this editPart

File: packages/mechanic-shop-fe/src/constants/actionTypes.js [hunk 4]
```diff
@@ -3,3 +3,6 @@ export const CURRENT_USER_FETCH = 'CURRENT_USER_FETCH';
 
 // Form submits
 export const LOGIN = 'LOGIN';
+
+// Add a new item
+export const ADD_ITEM = 'ADD_ITEM';
```
Comment (ghost) [inline packages/mechanic-shop-fe/src/constants/actionTypes.js]: Redux convention sais to use this as `noun(s)_verb` (verb at the end)

Please use the PART_ADD name

File: packages/mechanic-shop-fe/src/actions/index.js [hunk 5]
```diff
@@ -7,3 +7,5 @@ export const fetchCurrentUser = createAction(actionTypes.CURRENT_USER_FETCH);
 
 // Form submits & other actions altering backend state
 export const login = createAction(actionTypes.LOGIN);
+
+export const addItem = createAction(actionTypes.ADD_ITEM);
```
Comment (ghost) [inline packages/mechanic-shop-fe/src/actions/index.js]: Please use the name addPart

File: packages/mechanic-shop-fe/src/components/EditPartForm/index.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  SubmitButton,
+  FormError,
+} from '@autoquotes/common/src/components/Form';
+import TextInput from '@autoquotes/common/src/components/TextInput';
+import { Container, CssBaseline, Grid, Typography } from '@mui/material';
+
+const EditPartForm = props => {
+  return (
+    <Container sx={{ mt: 2, width: '80%' }}>
+      <CssBaseline />
```
Comment (ghost) [inline packages/mechanic-shop-fe/src/components/EditPartForm/index.js]: CssBaseLine tries to reset the css of external libraries. We don't have any external libraries messing with our CSS system, so this is not needed

File: packages/mechanic-shop-fe/src/paths.js [hunk 7]
```diff
@@ -4,6 +4,17 @@ const dashboard = () => ({
   pathname: '/dashboard',
 });
 
-const paths = { login, dashboard };
+const partList = () => ({
+  pathname: '/partList',
+});
+
+const addPartForm = () => ({
```
Comment (ghost) [inline packages/mechanic-shop-fe/src/paths.js]: Please rename to addPart

File: packages/mechanic-shop-fe/src/paths.js [hunk 8]
```diff
@@ -4,6 +4,17 @@ const dashboard = () => ({
...
+  pathname: '/addPartForm',
+});
+
+const editPartForm = () => ({
```
Comment (ghost) [inline packages/mechanic-shop-fe/src/paths.js]: please rename to editPart

**Review by ghost — CHANGES_REQUESTED** [review]
A few minor changes only.

File: packages/mechanic-shop-fe/src/Routes.js (hunk 1 above)
Comment (ghost) [reply packages/mechanic-shop-fe/src/Routes.js]: Thanks, I created some comments around naming

**Review by ghost — APPROVED** [review]
Very cool, hats off!

**Pull request merged**